tests/providers/docker/decorators/test_docker.py

```diff
@@ -60,6 +60,21 @@ def f(num_results):
         assert isinstance(result, list)
         assert len(result) == 50
 
+    def test_basic_docker_operator_with_template_fields(self, dag_maker):
+        @task.docker(image="python:3.9-slim", container_name='python_{{macros.datetime.now() | ts_nodash}}')
+        def f():
+            import random
+
+            return [random.random() for _ in range(100)]
+
+        with dag_maker():
+            ret = f()
+
+        dr = dag_maker.create_dagrun()
+        ret.operator.run(start_date=dr.execution_date, end_date=dr.execution_date)
+        ti = dr.get_task_instances()[0]
+        assert len(ti.xcom_pull()) == 100
+        
```

Comment:
We do not test here that python_{{macros.datetime.now() | ts_nodash}}` evaluated to `python_2018-01-01T00:00:00+00:00`

Reply:
But we evaluate that it actually run, which means that the JINJA has been processed, otherwise we would have "invalid name" (as in the original request). Good enough for me TBH

Reply:
```
Bad Request ("Invalid container name (python_{macros.datetime.now() | ts_nodash}), only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed")
```

Reply:
Some example how it done in `@task` decorator

https://github.com/apache/airflow/blob/884fca8d114ce8e0c982747937a1014f3b5e7491/tests/decorators/test_python.py#L238-L255

But please be aware that we never tested that default decorators attributes evaluated thought templated fields

Reply:
@potiuk @jeffery9876  testing rendered fields not required to run operator, so we could be sure that expected parameters pass to the operator.

When we run the operator, we actually check that operator could work with parameters. It is not our case but potentially it could be a situation that DockerOperator `slugify` parameters.

![image](https://user-images.githubusercontent.com/3998685/219637085-d49df489-7be5-4bb7-814a-db87df0843b1.png)

And in additional if we run `@task.docker(...)` without `auto_remove="force"` it would keep unwanted containers in dev system

![image](https://user-images.githubusercontent.com/3998685/219637766-1df26ba3-49ca-466c-b620-a3e50fada837.png)

Reply:
Thanks @Taragolis. I've removed the `ret.operator.run`, but I'm not 100% sure about it, hope it still works... :)